rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/condition/annotation/AnnotationTypeCondition.java

```diff
@@ -0,0 +1,80 @@
+package org.jboss.windup.rules.apps.java.condition.annotation;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.jboss.windup.config.GraphRewrite;
+import org.jboss.windup.config.condition.EvaluationStrategy;
+import org.jboss.windup.rules.apps.java.scan.ast.annotations.JavaAnnotationTypeReferenceModel;
+import org.jboss.windup.rules.apps.java.scan.ast.annotations.JavaAnnotationTypeValueModel;
+import org.ocpsoft.rewrite.context.EvaluationContext;
+import org.ocpsoft.rewrite.param.ParameterizedPatternResult;
+import org.ocpsoft.rewrite.param.RegexParameterizedPatternParser;
+
+/**
+ * @author <a href="mailto:jesse.sightler@gmail.com">Jesse Sightler</a>
+ */
+public class AnnotationTypeCondition extends AnnotationCondition
+{
```

Comment:
we probably need to add docs too in wiki. As this needs to be clear for end users how to use it properly.